reporter/gcppubsub/pubsub.go

```diff
@@ -0,0 +1,108 @@
+package gcppubsub
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/reporter"
+	"log"
+	"os"
+)
+
+const defaultPubSubTopic = "defaultTopic"
+
+// Reporter implements Reporter by publishing spans to a GCP gcppubsub.
+type Reporter struct {
+	logger *log.Logger
+	topic  string
+	client *pubsub.Client
+}
+
+// ReporterOption sets a parameter for the reporter
+type ReporterOption func(c *Reporter)
+
+// Send send span to topic
+func (r *Reporter) Send(s model.SpanModel) {
```

Comment:
I am curious about the tradeoff for reporing one span each time of piling up them up to a point and then do the reporting. I would prefer to not to do a http call on every span cc @basvanbeek

Reply:
We can for symmetry reasons with other reporters (and maybe optimization of message encoding/decoding). For request/response calls it is not needed as pubsub handles sending underneath and batches.... see https://godoc.org/cloud.google.com/go/pubsub